#187 Documents state migration (#188)
flink-extended/flink-scala-api: README.md

import org.apache.flinkx.api.serializers._
```

### State

Ensure to replace state descriptor constructors using `Class[T]` param with constructors using `TypeInformation[T]` or  `TypeSerializer[T]` param:

> ```scala

Reviewer: @arnaud-daroussin what if we make this small doc also validated by scala-mdoc?
It will require then to add class imports and perhaps case class definition.

Reviewer 2: It should be good.

Author: It should be good.

Reviewer: Ok, thank you

> `flink-scala-api` generates at compile-time specific TypeInformation & TypeSerializer. To use them for state serialization, ensure to replace state descriptor constructors using `Class[T]` param with constructors using `TypeInformation[T]` param:

Author: If you want to keep Migration section short, with explanations later in the readme, we can write this: 
```suggestion
Ensure to replace state descriptor constructors using `Class[T]` param with constructors using `TypeInformation[T]` or  `TypeSerializer[T]` param:
```
